app/controllers/report_controller/menus.rb

```diff
@@ -20,16 +20,17 @@ def menu_editor
     menu_set_form_vars if ["explorer", "tree_select", "x_history"].include?(params[:action])
     @in_a_form = true
     if @menu_lastaction != "menu_editor"
-      @menu_roles_tree = TreeBuilder.convert_bs_tree(build_menu_tree(@edit[:new])).to_json
+      @menu_roles_tree = TreeBuilderMenuRoles.new("menu_roles_tree", "menu_roles", @sb, session[:role_choice], @edit[:new])
```

Comment:
Also...

```
app/controllers/report_controller/menus.rb
13:      build_report_listnav("reports", "menu")
211:      @menu_roles_tree = TreeBuilder.convert_bs_tree(build_report_listnav("reports", "menu", "default")).to_json
```

These are the only places that use `hash_tree` outside of specs.

The second one you fixed here to use the TreeBuilder directly..
And the first one doesn't actually use the return value for anything :).

So you should be safely able to drop that whole `build_menu_tree` code path :).

As for using `hash_tree` itself .. please note that it's not such a good idea since the format of the data this generates will change soon (to essentially what `convert_bs_tree` outputs) .. so, maybe it'd be better to access these via the usual way so as not to introduce something different..